homeassistant/components/hassio/addon_panel.py

```diff
@@ -0,0 +1,93 @@
+"""Implement the Ingress Panel feature for Hass.io Add-ons."""
+import asyncio
+import logging
+
+from aiohttp import web
+
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import ATTR_PANELS, ATTR_TITLE, ATTR_ICON, ATTR_ADMIN, ATTR_ENABLE
+from .handler import HassioAPIError
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_addon_panel(hass: HomeAssistantType, hassio):
+    """Add-on Ingress Panel setup."""
+    hassio_addon_panel = HassIOAddonPanel(hass, hassio)
+    hass.http.register_view(hassio_addon_panel)
+
+    # If panels are exists
+    panels = await hassio_addon_panel.get_panels()
+    if not panels:
+        return
+
+    # Register available panels
+    jobs = []
+    for addon, data in panels.items():
+        if not data[ATTR_ENABLE]:
+            continue
+        jobs.append(_register_panel(hass, addon, data))
+
+    if jobs:
+        await asyncio.wait(jobs)
+
+
+class HassIOAddonPanel(HomeAssistantView):
+    """Hass.io view to handle base part."""
+
+    name = "api:hassio_push:panel"
+    url = "/api/hassio_push/panel/{addon}"
```

Comment:
From a REST perspective, wouldn't it make more sense to do `{addon}/panel` ? 

Also, what does `hassio_push` mean?

Reply:
oh nevermind about the hassio_push, that's for data pushed from hassio to hassio integration in hass.

Reply:
Yeah, the `hassio_push` is used by every view they trigger some things on Home Assistant 👍